dm/syncer/binlogstream/binlog_locations.go

```diff
@@ -52,41 +51,27 @@ type locationRecorder struct {
 	// - reset binlog replication for a finer granularity
 	// - save global checkpoint
 	txnEndLocation binlog.Location
+}
+
+// locationRecorder is not concurrent-safe.
+type locationRecorder struct {
+	locations
 
 	// DML will also generate a query event if user set session binlog_format='statement', we use this field to
 	// distinguish DML query event.
-	inDML bool
+	inDMLQuery bool
 
-	// we assign startGTID := endGTID after COMMIT, so at COMMIT we turn on the flag.
-	needUpdateStartGTID bool
-
-	mu sync.Mutex // guard curEndLocation because Syncer.printStatus is reading it from another goroutine.
+	preUpdate []func()
```

Comment:
It's better to define a new type for the `func()` to avoid abuse. For example: `type UpdateHookFunc func()`.
What do you think?

Reply:
https://github.com/pingcap/tiflow/pull/6489/commits/09e79391a5a713eb62bc10ee7f903f4a03b2d029